app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,37 @@
+class SessionsController < ApplicationController
+  def login_form; end
+
+  def new
+    @form_user = User.new
+  end
+
+# FIXME: Will not register me as logged in so I can vote
+  def login
+# **********
+    user_name = params[:user][:user_name]
+
+    if user_name and @user = User.find_by(user_name: user_name)
+      session[:user_id] = @user.id
+      flash[:result_text] = "Successfully logged in as exsisting user #{@user.user_name}"
+    else
+      @user = User.new(user_name: user_name)
+      if @user.save
+        session[:user_name] = @user.user_name
```

Comment:
Why are you storing the user's ID above and the name here? I think that this is why logging in will work sometimes but not all.